File: DosyaPaylasimJava/app/src/main/java/com/example/dosyapaylasimjava/service/RetrofitClient.java
package com.example.dosyapaylasimjava.service;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {

    private static final String BASE_URL = "https://alperensaracdeneme.com/api/";
    private static ApiService api;

    public static ApiService getApi() {
        if (api == null) {
            synchronized (RetrofitClient.class) {
                if (api == null) {
                    Retrofit retrofit = new Retrofit.Builder()
                            .baseUrl(BASE_URL)
                            .addConverterFactory(GsonConverterFactory.create())
                            .build();
                    api = retrofit.create(ApiService.class);
                }
            }
        }
        return api;
    }

    private RetrofitClient() {}
}

File: DosyaPaylasimJetpack/app/src/main/java/com/example/dosyapaylasimjetpack/service/RetrofitClient.kt
package com.example.dosyapaylasimjetpack.service

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClient {
    private const val BASE_URL = "https://alperensaracdeneme.com/api/"

    val api: ApiService by lazy {
        Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
            .create(ApiService::class.java)
    }
}
